packages/@lwc/template-compiler/src/codegen/helpers.ts

```diff
@@ -255,13 +279,18 @@ function collectStaticNodes(node: ChildNode, staticNodes: Set<ChildNode>, state:
     } else if (isComment(node)) {
         nodeIsStatic = true;
     } else {
-        // it is ForBlock | If | BaseElement
+        // it is ElseBlock | ForBlock | If | BaseElement
         node.children.forEach((childNode) => {
             collectStaticNodes(childNode, staticNodes, state);
 
             childrenAreStatic = childrenAreStatic && staticNodes.has(childNode);
         });
 
+        // for IfBlock and ElseifBlock, traverse down the else branch
+        if (isConditionalParentBlock(node) && node.else) {
```

Comment:
@jodarove Thanks for catching the issue with api_element vs api_static_fragment. It was missed when integrating with master.

I've updated the codegen logic here to traverse down the else branch of the conditional nodes like it should. The snapshots have also been updated.